src/core/Form/Label/Label.tsx

```diff
@@ -1,34 +1,54 @@
-import React, { Component, ReactNode } from 'react';
+import React, {
+  ReactElement,
+  SetStateAction,
+  Component,
+  ReactNode,
+  useState,
+  isValidElement,
+  cloneElement,
+  CSSProperties,
+} from 'react';
 import classnames from 'classnames';
 import { default as styled } from 'styled-components';
 import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../theme';
 import { baseStyles } from './Label.baseStyles';
 import { asPropType } from '../../../utils/typescript';
 import { VisuallyHidden } from '../../VisuallyHidden/VisuallyHidden';
-import { HtmlSpan, HtmlSpanProps, HtmlDiv, HtmlDivProps } from '../../../reset';
+import {
+  HtmlSpan,
+  HtmlSpanProps,
+  HtmlDivProps,
+  HtmlDivWithRef,
+} from '../../../reset';
 
 export type LabelMode = 'hidden' | 'visible';
 
-export interface LabelProps extends Omit<HtmlDivProps, 'as'> {
-  /** id */
+export interface LabelProps extends Omit<HtmlSpanProps, 'as'> {
+  /** id for label content */
   id?: string;
+  /** Wrapper class name for styling and customizing */
+  className?: string;
   /** Label element content */
   children: ReactNode;
-  /** Custom class name for styling and customizing  */
-  className?: string;
+  /** Content class name for styling and customizing */
+  contentClassName?: string;
+  /** Content style for styling and customizing */
+  style?: CSSProperties;
```

Comment:
As the `style` is for the content, should it be `contentStyle` to be consistent?

Reply:
Makes sense. In most cases the style prop is passed to the outermost element, and with that not being the case here, it would make sense to communicate that.

Reply:
This is now changed.